docs/developers_notes/03-base_regressor.md

```diff
@@ -0,0 +1,59 @@
+# The Abstract Class `BaseRegressor`
+
+`BaseRegressor` is an abstract class that inherits from `Base`, stipulating the implementation of number of abstract methods such as `fit`, `predict`, `score`. This ensures seamless assimilation with `scikit-learn` pipelines and cross-validation procedures.
+
+
+!!! Example
+    The current package version includes a concrete class named `nemos.glm.GLM`. This class inherits from `BaseRegressor`, which in turn inherits `Base`, since it falls under the " GLM regression" category. 
+    As any `BaseRegressor`, it **must** implement the `fit`, `score`, `predict` methods.
+
+### Abstract Methods
+
+For subclasses derived from `BaseRegressor` to function correctly, they must implement the following:
+
+1. `fit`: Adapt the model using input data `X` and corresponding observations `y`.
+2. `predict`: Provide predictions based on the trained model and input data `X`.
+3. `score`: Score the accuracy of model predictions using input data `X` against the actual observations `y`.
+4. `simulate`: Simulate data based on the trained regression model.
+5. `update`: Run a single optimization step, and stores the updated parameter and solver state. Used by stochastic optimization schemes.
+6. `_predict_and_compute_loss`: Compute prediction and evaluates the loss function prvided the parameters and `X` and `y`.
+7. `_check_params`: Check the parameter structure.
+8. `_check_input_dimensionality`: Check the input dimensionality matches model expectation.
+9. `_check_input_and_params_consistency`: Checks that the input and the parameters are consistent.
+10. `_get_coef_and_intercept` and `_set_coef_and_intercept`: set and get model coefficient and intercept term.
+
+
+### Attributes
+
+Public attributes are stored as properties:
```

Comment:
worth saying: when writing a new baseregressor, do you need to do anything with these or do they come along for free?

Reply:
I added a comment